### DISCUSSION TITLE
React native library

### DISCUSSION THREAD

**Discussion by jhotadhari** [mapsforge/vtm#1147] [Show and tell]
Hi, I started a library for react-native. It's quite limited yet and just has some basic features.

Maybe it's helpful for someone.

More information here:
https://github.com/jhotadhari/react-native-mapsforge-vtm
https://www.npmjs.com/package/react-native-mapsforge-vtm

- Component for online xyz raster tile layer.
- Mapsforge layer with support for external xml render styles.
- MBtiles bitmap layer 
- Support for HGT offline hillshading, https://github.com/mapsforge/vtm/issues/1101
That's the [HillshadingTileSource](https://github.com/jhotadhari/react-native-mapsforge-vtm/blob/main/android/src/main/java/com/jhotadhari/reactnative/mapsforge/vtm/tiling/source/hills/HillshadingTileSource.java)
- Elevation at map center from hgt offline dem.
- A path component that renders the colors accordingly to the slope, https://github.com/mapsforge/vtm/discussions/868
- A path component that uses the vtm jts `org.oscim.layers.vector.PathLayer` https://github.com/mapsforge/vtm/discussions/1139
- no markers yet, and lots of stuff missing ...

Some [example application](https://github.com/jhotadhari/react-native-mapsforge-vtm/tree/main/example) screenshots:
![image](https://github.com/user-attachments/assets/39e133ae-2283-441d-af29-ab62e39e405b)
![image](https://github.com/user-attachments/assets/c697c95c-24ab-4837-b837-f0d2eb819a62)
![image](https://github.com/user-attachments/assets/357fe3ce-7c9b-4990-8c5b-2d56c3f86814)
![image](https://github.com/user-attachments/assets/2295b084-9916-4a21-a3fa-5fd37eb3704f)
![image](https://github.com/user-attachments/assets/74b82e7a-ce5c-4278-8be7-e8437140be60)

**Comment by devemux86** [comment]
OpenGL has many problems, rendering artifacts and even [fails to run](https://github.com/mapsforge/vtm/issues/1125).

Mapsforge is the recommended map library, now with **map rotation** and new hill shading:
- https://github.com/mapsforge/mapsforge/issues/7#issuecomment-2147721066
- https://github.com/mapsforge/mapsforge/issues/1551

**Reply by demiantres** [reply]
I prefer VTM - it just looks better and feels much faster (in most situations).
